File: src/main/java/br/com/guizzo/projectbasic/modules/storedfile/domain/entities/StoredFile.java
package br.com.guizzo.projectbasic.modules.storedfile.domain.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.Date;
import java.util.UUID;

@Entity
@Table(name = "stored_files")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class StoredFile implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "uuid", length = 36, unique = true, nullable = false)
    private String uuid;

    @Column(nullable = false, length = 100)
    private String alt;

    @Column(name = "original_name", nullable = false, length = 100)
    private String originalName;

    @Column(name = "stored_name", nullable = false, length = 100)
    private String storedName;

    @Column(name = "relative_path", nullable = false, length = 100)
    private String relativePath;

    @Column(name = "content_type", nullable = false, length = 100)
    private String contentType;

    @Column(name = "is_public", nullable = false)
    private Boolean isPublic;

    @Column(nullable = false, length = 100)
    private String checksum;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "created_at", nullable = false, updatable = false)
    private Date createdAt;

    @PrePersist
    protected void onCreate() {
        this.uuid = UUID.randomUUID().toString();
        this.createdAt = new Date();
    }
}

File: src/modules/stored_file/domain/entities/stored_file.py
import uuid
from django.db import models


class StoredFile(models.Model):
    id = models.AutoField(primary_key=True)
    uuid = models.UUIDField(unique=True, default=uuid.uuid4, editable=False)
    alt = models.CharField(max_length=255)
    original_name = models.CharField(max_length=255)
    stored_name = models.CharField(max_length=255)
    relative_path = models.CharField(max_length=255)
    content_type = models.CharField(max_length=100)
    is_public = models.BooleanField(default=False)
    checksum = models.CharField(max_length=255)
    created_at = models.DateTimeField(auto_now_add=True)

    class Meta:
        db_table = "stored_files"

    def __str__(self):
        return self.stored_name
